scripts/utils.py

```diff
@@ -138,7 +140,82 @@ def svg_preprocess(inputs: Dict, preprocess: Callable):
         base64_str = str(encoded_string, "utf-8")
         base64_str = "data:image/png;base64," + base64_str
         inputs["image"] = base64_str
-    return preprocess(inputs)
+    return inputs
+
+
+# This is a patch for `gr.Image.preprocess()`
+def image_preprocess(self, x: Dict):
+    inputs = x # lackluster name `x` is the method keyword name used by `gr.Image.preprocess`
+    inputs = svg_preprocess(inputs)
+
+    if self.image_mode is None:
+        # XXX(teding): Not sure if this early return is necessary, since I don't know why the caller is using `self.image_mode=None`.
+        # If 16-bit grayscale images are getting incorrectly converted to 8-bit, try removing this early return.
+        ret = gr.Image.preprocess(self, inputs)
+        return ret
+
+    # Work around for 16 bit grayscale images being improperly converted to 8 bit
+    image_mode = self.image_mode
+    image_type = self.type
+    if self.image_mode not in [None, "RGB"]:
+        raise Exception(f"Unsupported image mode {self.image_mode}")
+    if self.type not in ["numpy", "pil"]:
+        raise Exception(f"Unsupported image type {self.type}")
+    self.image_mode = None
+    self.type = "pil"
+
+    ret = gr.Image.preprocess(self, inputs)
+
+    def go_image(image, input):
+        if image is None:
+            return None
+        # gr.Image.preprocess() throws away depth information, so we need to decode again.
+        # Naively just checking if any elem is greater than 255 is not enough, because
+        # the image may be a (very dark) 16 bit grayscale image with all values below 256.
+        png_encoded = gr.processing_utils.extract_base64_data(input)
+        png = Image.open(io.BytesIO(base64.b64decode(png_encoded)))
+        assert len(png.tile) == 1
```

Comment:
I should probably make this an `if` instead of an assert, in case the user sends an image stored with multiple tiles (probably very rare?). In that case, it's probably better that the image gets decoded without this scaling in case some of the other tiles don't correspond to the same depth as `tile[0]`

Either that or I learn how to handle tiles and scale the data appropriately. Might be as simple as sliding non-overlapping windows... which is quite doable.